hash.c

```diff
@@ -4304,45 +4336,75 @@ rb_hash_update_block_i(VALUE key, VALUE value, VALUE hash)
 
 /*
  *  call-seq:
- *     hsh.merge!(other_hash1, other_hash2, ...)              -> hsh
- *     hsh.update(other_hash1, other_hash2, ...)              -> hsh
- *     hsh.merge!(other_hash1, other_hash2, ...) {|key, oldval, newval| block}
- *                                                            -> hsh
- *     hsh.update(other_hash1, other_hash2, ...) {|key, oldval, newval| block}
- *                                                            -> hsh
- *
- *  Adds the contents of the given hashes to the receiver.
- *
- *  If no block is given, entries with duplicate keys are overwritten
- *  with the values from each +other_hash+ successively,
- *  otherwise the value for each duplicate key is determined by
- *  calling the block with the key, its value in the receiver and
- *  its value in each +other_hash+.
- *
- *     h1 = { "a" => 100, "b" => 200 }
- *     h1.merge!          #=> {"a"=>100, "b"=>200}
- *     h1                 #=> {"a"=>100, "b"=>200}
- *
- *     h1 = { "a" => 100, "b" => 200 }
- *     h2 = { "b" => 246, "c" => 300 }
- *     h1.merge!(h2)      #=> {"a"=>100, "b"=>246, "c"=>300}
- *     h1                 #=> {"a"=>100, "b"=>246, "c"=>300}
- *
- *     h1 = { "a" => 100, "b" => 200 }
- *     h2 = { "b" => 246, "c" => 300 }
- *     h3 = { "b" => 357, "d" => 400 }
- *     h1.merge!(h2, h3)
- *                        #=> {"a"=>100, "b"=>357, "c"=>300, "d"=>400}
- *     h1                 #=> {"a"=>100, "b"=>357, "c"=>300, "d"=>400}
- *
- *     h1 = { "a" => 100, "b" => 200 }
- *     h2 = { "b" => 246, "c" => 300 }
- *     h3 = { "b" => 357, "d" => 400 }
- *     h1.merge!(h2, h3) {|key, v1, v2| v1 }
- *                        #=> {"a"=>100, "b"=>200, "c"=>300, "d"=>400}
- *     h1                 #=> {"a"=>100, "b"=>200, "c"=>300, "d"=>400}
- *
- *  Hash#update is an alias for Hash#merge!.
+ *    hash.merge! -> self
+ *    hash.merge!(*other_hashes) -> self
+ *    hash.merge!(*other_hashes) { |key, old_value, new_value| ... } -> self
```

Comment:
We are missing a one line synopsis.

This method is very important to document clearly.

Reply:
Fixed.